ir/ir.rs

```diff
@@ -468,6 +468,22 @@ impl Item {
 
     /// Get this item's name.
     pub fn name(&self) -> String {
+        match &self.kind {
+            ItemKind::Code(code) => code
+                .demangled()
+                .or_else(|| code.name())
+                .unwrap_or_else(|| code.decorator())
+                .to_string(),
+            ItemKind::Data(Data { name, .. }) => name.to_string(),
+            ItemKind::Func(func) => func.decorator().to_string(),
+            ItemKind::Debug(DebugInfo { name, .. }) => name.to_string(),
+            ItemKind::Misc(Misc { name, .. }) => name.to_string(),
+        }
+    }
+
+    /// Get this item's decorated name.
+    /// TODO: This needs better documentation.
```

Comment:
Yes, please :)

Reply:
Ditto here, the `TODO` was removed, but I still have the same sentiment. Maybe a simple example, something like:

```
/// Get this item's decorated name, i.e. 'foo (code[7])'
```

That might warrant adding some other info to the existing `name` method, which I'm also happy to add if you think that makes sense :slightly_smiling_face: